src/patient-card/patient-card-component.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+import { Button } from "carbon-components-react";
+import styles from "./patient-card.scss";
+import { ConfigurableLink, ExtensionSlot } from "@openmrs/esm-framework";
+import CaretDown16 from "@carbon/icons-react/es/caret--down/16";
+import CaretUp16 from "@carbon/icons-react/es/caret--up/16";
+import ContactDetails from "../contact-details/contact-details.component";
+
+export interface PatientCardProps {
+  patientUuid: string;
+  displayName: string;
+  gender: string;
+  birthDate: string;
+  identifier: string;
+  age: string;
+  address: fhir.Address[];
+  telecom: fhir.ContactPoint[];
+  patientUrl?: string;
+}
+
+const PatientCard: React.FC<PatientCardProps> = ({
+  patientUuid,
+  displayName,
+  gender,
+  birthDate,
+  identifier,
+  age,
+  address,
+  telecom,
+  patientUrl
+}): JSX.Element => {
+  const [showContactDetails, setShowContactDetails] = React.useState(false);
+  const toggleContactDetails = () => {
+    setShowContactDetails(!showContactDetails);
+  };
+
+  const patientAvatar = (
+    <div className={styles.patientAvatar}>
+      <ExtensionSlot
+        extensionSlotName="patient-photo"
+        state={{ patientUuid: patientUuid }}
```

Comment:
You are recreating the state on each render. Please memo it using `patientUuid` as dependency.

Reply:
Done... check my new commits